### PR TITLE
Added support for destinations

### PR DESCRIPTION
**Pull request by mathew-tuvesson-cko**

### CODE REVIEW

File: spec/components/schemas/Payments/Payment.yaml [hunk 1]
```diff
@@ -67,6 +72,7 @@ properties:
       - Refunded
       - Declined
       - Cancelled
+      - Paid
```
Comment (ben-foster-cko) [inline spec/components/schemas/Payments/Payment.yaml]: To be discussed

**Comment by ben-foster-cko** [comment]
@francesca-gaunard-cko please can you review the copy? This should not be merged until we've released the feature.

File: spec/paths/payments.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ post:
   summary: Request a payment
   description: |
     Checkout.com supports payments from a variety of sources, from cards to
-    SEPA direct debits. The source of payment must be specified using the `source.type` field along with any source specific data.
+    SEPA direct debits, to a variety of destinations. The source or destination of payment must be specified using the `source.type` or `destination.type` field along with any source or destination specific data.
```
Comment (toby-knott-cko) [inline spec/paths/payments.yaml]:
```suggestion
Accept payments from <a href="https://docs.checkout.com/docs/cards" target="blank">cards</a>, <a href="https://docs.checkout.com/docs/wallets" target="blank">digital wallets</a> and many <a href="https://docs.checkout.com/docs/alternative-payments" target="blank">alternative payment methods</a>, and pay out to a variety of destinations. Just specify the source or destination of the payment using the `source.type` or `destination.type` field, along with the source- or destination-specific data.
```

File: spec/components/schemas/Payments/ResponseDestinations/01_PaymentResponseCardDestination.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        type: string
+        description: The card scheme
+        example: "VISA"
+      last4:
+        type: string
+        description: The last four digits of the card number
+        example: "9996"
+      fingerprint:
+        type: string
+        description: Uniquely identifies this particular card number. You can use this to compare cards across customers
```
Comment (toby-knott-cko) [inline spec/components/schemas/Payments/ResponseDestinations/01_PaymentResponseCardDestination.yaml]: Should be a full stop after 'customers'.

File: spec/components/schemas/Payments/ResponseDestinations/01_PaymentResponseCardDestination.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        example: "F639CAB2745BEE4140BF86DF6B6D6"
+      bin:
+        type: string
+        description: The card issuer BIN
```
Comment (toby-knott-cko) [inline spec/components/schemas/Payments/ResponseDestinations/01_PaymentResponseCardDestination.yaml]:
```suggestion
        description: The card issuer's BIN
```

File: spec/components/schemas/Payments/ResponseDestinations/01_PaymentResponseCardDestination.yaml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        example: Consumer
+      issuer:
+        type: string
+        description: The name of the card issuer
+        example: "GOTHAM STATE BANK"
+      issuer_country:
+        type: string
+        maxLength: 2
+        minLength: 2
+        description: The card issuer country ISO-2 code
```
Comment (toby-knott-cko) [inline spec/components/schemas/Payments/ResponseDestinations/01_PaymentResponseCardDestination.yaml]:
```suggestion
        description: The card issuer's country (<a href="https://docs.checkout.com/docs/country-codes" target="blank">two-letter ISO code</a>)
```

File: spec/components/schemas/Payments/RequestDestinations/01_PaymentRequestTokenDestination.yaml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  - token
+  - first_name
+  - last_name
+allOf:
+  - $ref: "#/components/schemas/PaymentRequestDestination"
+  - type: object
+    properties:
+      token:
+        type: string
+        description: The Checkout.com token for example a card or wallet or token
```
Comment (toby-knott-cko) [inline spec/components/schemas/Payments/RequestDestinations/01_PaymentRequestTokenDestination.yaml]:
```suggestion
        description: The Checkout.com token (e.g., a card, wallet or token)
```

File: spec/components/schemas/Payments/RequestDestinations/01_PaymentRequestTokenDestination.yaml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+      first_name:
+        type: string
+        description: The payment destination owner's first name
+        example: John
+      last_name:
+        type: string
+        description: The payment destination owner's last name
+        example: Smith
+      billing_address:
+        description: The payment destination owner's billing address. This will override the billing address specified during tokenisation
```
Comment (toby-knott-cko) [inline spec/components/schemas/Payments/RequestDestinations/01_PaymentRequestTokenDestination.yaml]: Should be a full stop after 'tokenisation'.

File: spec/components/schemas/Payments/RequestDestinations/01_PaymentRequestTokenDestination.yaml [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        allOf:
+          - $ref: '#/components/schemas/Address'
+      phone:
+        description: The payment destination owner's phone number. This will override the phone number specified during tokenisation
```
Comment (toby-knott-cko) [inline spec/components/schemas/Payments/RequestDestinations/01_PaymentRequestTokenDestination.yaml]: Should be a full stop after 'tokenisation'.

File: spec/components/schemas/Payments/RequestDestinations/02_PaymentRequestIdDestination.yaml [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+  - id
+  - first_name
+  - last_name
+allOf:
+  - $ref: "#/components/schemas/PaymentRequestDestination"
+  - type: object
+    properties:
+      id:
+        type: string
+        description: The payment source identifer for example, a card source identifier
```
Comment (toby-knott-cko) [inline spec/components/schemas/Payments/RequestDestinations/02_PaymentRequestIdDestination.yaml]:
```suggestion
        description: The payment source identifier (e.g., a card source identifier)
```

File: spec/components/schemas/Payments/RequestDestinations/03_PaymentRequestCardDestination.yaml [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+        type: string
+        description: The payment destination owner's first name
+        example: John
+      last_name:
+        type: string
+        description: The payment destination owner's last name
+        example: Smith
+      name:
+        type: string
+        description: The card-holder name
```
Comment (toby-knott-cko) [inline spec/components/schemas/Payments/RequestDestinations/03_PaymentRequestCardDestination.yaml]:
```suggestion
        description: The cardholder's name
```

File: spec/components/schemas/Payments/ResponseDestinations/01_PaymentResponseCardDestination.yaml [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+        description: The two-digit expiry month
+        minimum: 1
+        example: 6
+      expiry_year:
+        type: integer
+        description: The four-digit expiry year
+        example: 2025
+      name:
+        type: string
+        description: The card-holder name
```
Comment (toby-knott-cko) [inline spec/components/schemas/Payments/ResponseDestinations/01_PaymentResponseCardDestination.yaml]:
```suggestion
        description: The cardholder's name
```

**Pull request merged**